### ISSUE TITLE
render_to_span

### ISSUE THREAD

**Issue by danniranderis** [mitya57/python-markdown-math#15]
Hi.

I can see that the config-option "Render_to_span" making it easier to support users that has disabled js is committed 13 months ago.
Would it be possible to ship a new release anytime soon, so that (large performance change) can get out into the air?

Thanks in advance!
Danni

**Comment by mitya57** [comment]
I can make a new release of course.

Are you satisfied with the current behavior of that option? It currently renders only inline math to spans, for standalone equations it still uses `<script>` tags.

**Comment by danniranderis** [comment]
For my usages inline is enough, but for sure - standalone should be processed the same way.

A thing I have noticed - and it might me my lack of enough understanding of configuring MathJax, is that this right now creates a "hard-fallback" where MathJax is not rendering the equations if javascript is available due to the fact that it looks for `<script type="math/tex">` and not `<span class="tex">`. As of right now - I haven't figured out, how to solve this.

**Comment by danniranderis** [comment]
[config-hub](https://docs.mathjax.org/en/v2.5-latest/options/hub.html#configure-hub) --> preRemoveClass: "MathJax_Preview".

I think we might use this to our advantage to achieve both no-js and js.

While not converting `<script>` to `<span>` it would rather be better to just add an extra `<span class="MathJax_Preview">` before the `<script>`-tag.

With this approach we will get the following: 1) with no js, the `<span>` is not removed and therefore visible and the `<script>` is of course invisible. 2) with js enabled, the `<span>` is removed and `<script>` is processed.

**Comment by danniranderis** [comment]
I have now forked this and created a solution for the approach described above. Feel free to test it - I haven't seen any errors with it and it does not need extra config for MathJax.

**Comment by mitya57** [comment]
Now I want to check with the author of the original change. @a3nm, are you OK with the proposed change?

**Comment by danniranderis** [comment]
Sounds good - I have also added the same function to `handle_match()` so that standalone equations are handled the same way when `render_to_span=True`.

Looking forward to hear from you

**Comment by a3nm** [comment]
I'm not sure I remember what all of this is about :) but I just pulled
the proposed changes and they don't seem to introduce any issues for
what I do, so I'm fine with the change. Thanks for letting me know.

-- 
Antoine Amarilli

**Comment by mitya57** [comment]
I have simplified the code, added tests, and renamed the option from `render_to_span` to `add_preview` (now that it outputs not only `span`/`div`, but also the `script` tag).

Please test it and I will roll a new release.

**Comment by a3nm** [comment]
This appears to work fine for me. (Actually my use case is to pass the
result to MathJax-node/bin/page2html to render the MathJax server-side,
so I don't need the extra <script>, but it appears that none of this
remains in the output of page2.html.)

-- 
Antoine Amarilli

**Comment by danniranderis** [comment]
Sorry for not looking on this before now..

I have now tested your commit on my setup and it works perfect. I like the renaming - it makes more sense :-)
And nice simplification on the provided code! I love, when I can learn a little bit every day. Thanks for that :-)

As far as I can tell, this should be ready for rolling a new release.

**Comment by mitya57** [comment]
Sorry for the delay too. The new release is pushed now.
